core/lib/spree/core/controller_helpers/order.rb

```diff
@@ -52,6 +52,8 @@ def current_order(options = {})
         end
 
         def associate_user
+          return if Spree::Config.manually_associate_user
+          warn "Order#associate_user is deprecated and will be removed in 2.11"
```

Comment:
Great! All deprecation in Solidus are managed via `Spree::Deprecation.warn` (see examples around the code) and would also be better to have a spec ensuring the deprecation is displayed (there are numerous examples of that in spec/ as well).